app/src/androidTest/kotlin/at/bitfire/davdroid/db/AppDatabaseMigrationsTest.kt

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright © All Contributors. See LICENSE and AUTHORS in the root directory for details.
+ */
+
+package at.bitfire.davdroid.db
+
+import android.content.Context
+import androidx.room.Room
+import androidx.room.testing.MigrationTestHelper
+import androidx.sqlite.db.SupportSQLiteDatabase
+import androidx.sqlite.db.framework.FrameworkSQLiteOpenHelperFactory
+import androidx.test.platform.app.InstrumentationRegistry
+import at.bitfire.davdroid.db.Collection.Companion.TYPE_CALENDAR
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertNotNull
+import org.junit.Assert.assertNull
+import org.junit.Rule
+import org.junit.Test
+
+class AppDatabaseMigrationsTest {
+
+    private val TEST_DB = "test"
+
+    val context: Context = InstrumentationRegistry.getInstrumentation().targetContext
+
+    private val autoMigrationSpecs = AppDatabase.autoMigrationSpecs.map { it(context) }
+
+    @Rule
+    @JvmField
+    val helper = MigrationTestHelper(
+        InstrumentationRegistry.getInstrumentation(),
+        AppDatabase::class.java,
+        autoMigrationSpecs,
+        FrameworkSQLiteOpenHelperFactory()
+    )
+
+    /**
+     * Used for testing the migration process between two versions.
+     * @param fromVersion The version from which to start testing.
+     * @param toVersion The target version to test
+     * @param setup Run here all the SQL queries that prepare the database with the data required
+     * to perform the migration process.
+     * @param block Run here all the assertions for [AppDatabase] to make sure the migration was
+     * performed correctly.
+     */
+    private fun testMigration(
+        fromVersion: Int,
+        toVersion: Int,
+        setup: SupportSQLiteDatabase.() -> Unit,
+        block: (AppDatabase) -> Unit
```

Comment:
Would probably call this param something with "assert" like `assertionsBlock` or similar